### ISSUE TITLE
`melt()`: extract multiple variables from name

### ISSUE THREAD

**Issue by mgirlich** [pola-rs/polars#15276]
### Description

In the following example are two columns whose name contain multiple variables. To extract the variables (and sort the columns in a better order) the following code is needed at the moment:

```python
who = pl.from_repr("""
┌──────┬──────┬─────────────┬──────────────┐
│ iso2 ┆ year ┆ new_sp_f014 ┆ new_sp_f1524 │
│ ---  ┆ ---  ┆ ---         ┆ ---          │
│ str  ┆ i64  ┆ i64         ┆ i64          │
╞══════╪══════╪═════════════╪══════════════╡
│ ZW   ┆ 2009 ┆ 180         ┆ 873          │
│ ZW   ┆ 2010 ┆ 173         ┆ 974          │
│ ZW   ┆ 2011 ┆ 174         ┆ 1084         │
│ ZW   ┆ 2012 ┆ 173         ┆ 939          │
│ ZW   ┆ 2013 ┆ null        ┆ null         │
└──────┴──────┴─────────────┴──────────────┘
""")
    
who.melt(
    id_vars=["iso2", "year"],
    value_name="count"
).with_columns(
    variable=pl.col("variable").str.extract_groups("new_?(?<diagnosis>.*)_(?<gender>.)(?<age>.*)")
).unnest("variable").select(
    "iso2",
    "year",
    "diagnosis",
    "gender",
    "age",
    "count",
)
```

It would be nice to shorten this to

```python
who.melt(
    id_vars=["iso2", "year"],
    value_name="count",
    variable_pattern="new_?(?<diagnosis>.*)_(?<gender>.)(?<age>.*)",
)

# or alternatively

who.melt(
    id_vars=["iso2", "year"],
    value_name="count",
    variable_pattern="new_?(.*)_(.)(.*)",
    variable_name=["diagnosis", "gender", "age"],
)
```

It would probably make sense to change the singular `variable_name` to the plural `variable_names`.

(This feature is basically what the R function `tidyr::pivot_longer()` can do. You can find some more background on this functionality in [this vignette](https://tidyr.tidyverse.org/articles/pivot.html#many-variables-in-column-names))

**Comment by avimallu** [comment]
Broadly a duplicate of https://github.com/pola-rs/polars/issues/8903.
